weave/compile.py

```diff
@@ -87,6 +87,39 @@ def _quote_node(node: graph.Node) -> graph.Node:
     return weave_internal.const(node)
 
 
+def _node_simplification(node: graph.Node) -> typing.Optional[graph.OutputNode]:
```

Comment:
There is another pass called "compile_simple_optimizations" that does stuff kinda like this. I wonder if they should be merged.

Reply:
done in next commit